### PR TITLE
Design Reviews: Initial diagrams and ideas on high level design recipe

### PR DESCRIPTION
**Pull request by ScoGroMSFT**
- Initial cut at high level design criteria
- Added example sequence + box diagrams
- Fixed typo in the milestone/epic recipe

#133

### CODE REVIEW

File: design-reviews/recipes/high-level-design-recipe.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# High Level / Game Plan Design Recipe
+
+## Why is this valuable?
+
+Design at macroscopic level shows the interactions between systems and services that will be used to accomplish the project. It is intended to ensure there is high level understanding of the plan for what to build, which off-the-shelf components will be used, and which external components will need to interact with the deliverable.
+
+## Things to keep in mind
```
Comment (wbreza) [inline design-reviews/recipes/high-level-design-recipe.md:7]: One thing we should consider adding is attempt to illustrate the different personas involved in the use cases and how/which boxes are their entrypoints.

Comment (wbreza) [reply design-reviews/recipes/high-level-design-recipe.md:7]: Some other ideas could be around existing vs. net new components.  I saw some colors illustrated in your diagram but it wasn't clear if that was referring to this concept or not.

Comment (ScoGroMSFT) [reply design-reviews/recipes/high-level-design-recipe.md:7]: Great feedback! Incorporated into the change

File: design-reviews/recipes/high-level-design-recipe.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+* As with all other aspects of the project, design reviews must provide a friendly and safe environment so that any team member feels comfortable proposing a design for review and can use the opportunity to grow and learn from the constructive / non-judgemental feedback from peers and subject matter experts (see [Team Agreements](../../team-agreements)).
+* Attempt to illustrate different personas involved in the use cases and how/which boxes are their entrypoints.
+* Prefer pictures over paragraphs. The diagrams aren't intended to generate code so they should be fairly high level.
+* Artifacts should indicate the direction of calls (are they outbound, inbound, or bidirectional?) and call out system boundaries where ports might need to be opened or additional infrastructure work may be needed to allow calls to be made.
+* Sequence diagrams are helpful to show the flow of calls among components + systems.
+* Generic box diagrams depicting data flow or call origination/destination are useful. However, the title should clealy define what the arrows show indicate. In most cases, a diagram will show either data flow or call directions but not both.
```
Comment (alkofahi) [inline design-reviews/recipes/high-level-design-recipe.md:14]: "should clealy define" -> missing an r

File: design-reviews/recipes/high-level-design-recipe.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* Clearly indicate existing vs. new components (color or grouping boxes can be used to differentiate this.)
```
Comment (alkofahi) [inline design-reviews/recipes/high-level-design-recipe.md:15]: How about rewording this, something along the lines of "visualize the different aspects of the system/diagram for ease of communication"? A diagram can be rich in info, for instance, what tech will be used, components that will be modified vs remaining untouched, boundaries (i.e. internet vs local vs cloud...etc),...etc. Here is an example

image

File: design-reviews/recipes/high-level-design-recipe.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+* Prefer ease-of-understanding for communicating ideas over strict UML correctness.
```
Comment (alkofahi) [inline design-reviews/recipes/high-level-design-recipe.md:16]: Should these points be sub-points to line 11 "Prefer pictures over paragraphs"?

File: design-reviews/recipes/high-level-design-recipe.md (hunk 3 above)
Comment (alkofahi) [reply design-reviews/recipes/high-level-design-recipe.md:15]: I think GitHub did not like me adding overlapping comments! This comment is specifically for line 15 (Clearly indicate...etc)

**Comment by fnocera** [comment]
@ScoGroMSFT please address the comments and then @alkofahi can re-review - hoping we can get this finalized and merged asap. Thanks!! 

cc @wbreza

File: design-reviews/recipes/high-level-design-recipe.md (hunk 4 above)
Comment (ScoGroMSFT) [reply design-reviews/recipes/high-level-design-recipe.md]: Good call! Fixed

File: design-reviews/recipes/high-level-design-recipe.md (hunk 3 above)
Comment (ScoGroMSFT) [reply design-reviews/recipes/high-level-design-recipe.md]: I took a stab at this. However, I like the diagram that you used to illustrate. Can I include that as an example?

File: design-reviews/recipes/high-level-design-recipe.md (hunk 2 above)
Comment (ScoGroMSFT) [reply design-reviews/recipes/high-level-design-recipe.md]: Fixed

File: design-reviews/recipes/high-level-design-recipe.md (hunk 3 above)
Comment (alkofahi) [reply design-reviews/recipes/high-level-design-recipe.md]: Sure thing, let me know if you would like to make some edits and I can share the power point.

**Review by alkofahi — APPROVED** [review]
LGTM
